packages/mdc-segmented-button/segment/foundation.ts

```diff
@@ -42,22 +42,37 @@ export class MDCSegmentedButtonSegmentFoundation extends MDCFoundation<MDCSegmen
   }
 
   isSelected(): boolean {
-    return false;
+    return this.adapter.hasClass('mdc-segmented-button__segment--selected');
   }
 
   setSelected() {
-    return;
+    this.adapter.addClass('mdc-segmented-button__segment--selected');
+    this.setAriaAttr('true');
   }
 
   setUnselected() {
-    return;
+    this.adapter.removeClass('mdc-segmented-button__segment--selected');
+    this.setAriaAttr('false');
   }
 
   getSegmentId(): string {
-    return '';
+    return this.adapter.getAttr('id');
```

Comment:
Are we not using data attribute instead of HTML `id` attribute for segment id?

Reply:
Oh yeah, I meant to put `segmentId` instead of `id`. But I wasn't familiar with the difference between a data attribute and a regular attribute when I wrote this. Do you think it's ok to have the Component handle the difference when implementing the Adapter, or should there be a Adapter.getCustomAttr or Adapter.getDataAttr method?

Reply:
I think foundation should be opinionated about which attribute name to fetch. It can use `data-segment-id` directly instead of `id`.